src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,210 @@
+		syntaxJsonObject.add("patterns", convertToJsonArray(eventInfo.patterns));
+		return syntaxJsonObject;
+	}
+
+
+	private <T extends StructureInfo<? extends Structure>> JsonArray generateStructureElementArray(Iterator<T> infos) {
+		JsonArray syntaxArray = new JsonArray();
+		infos.forEachRemaining(info -> {
+			if (info instanceof SkriptEventInfo) {
+				syntaxArray.add(generateEventElement((SkriptEventInfo<?>) info));
+			} else {
+				JsonObject structureElementJsonObject = generatedAnnotatedElement(info.getElementClass(), info.patterns);
+				if (structureElementJsonObject != null)
+					syntaxArray.add(structureElementJsonObject);
+			}
+		});
+		return syntaxArray;
+	}
+
+	private <T extends SyntaxElementInfo<? extends SyntaxElement>> JsonArray generateSyntaxElementArray(Iterator<T> infos) {
+		JsonArray syntaxArray = new JsonArray();
+		infos.forEachRemaining(info -> {
+			JsonObject syntaxJsonObject = generatedAnnotatedElement(info.getElementClass(), info.patterns);
+			if (syntaxJsonObject != null)
+				syntaxArray.add(syntaxJsonObject);
+		});
+		return syntaxArray;
+	}
+
+	private @Nullable JsonObject generateClassInfoElement(ClassInfo<?> classInfo) {
+		if (!classInfo.hasDocs())
+			return null;
+		JsonObject syntaxJsonObject = new JsonObject();
+		syntaxJsonObject.addProperty("id", idProvider.getId(classInfo));
+		syntaxJsonObject.addProperty("name", classInfo.getDocName());
+		syntaxJsonObject.addProperty("since", classInfo.getSince());
+		syntaxJsonObject.add("description", convertToJsonArray(classInfo.getDescription()));
+		syntaxJsonObject.add("examples", convertToJsonArray(classInfo.getExamples()));
+		syntaxJsonObject.add("patterns", convertToJsonArray(classInfo.getUsage()));
+		return syntaxJsonObject;
+	}
+
+	private JsonArray generateClassInfoArray(Iterator<ClassInfo<?>> classInfos) {
+		JsonArray syntaxArray = new JsonArray();
+		classInfos.forEachRemaining(classInfo -> {
+			JsonObject classInfoElement = generateClassInfoElement(classInfo);
+			if (classInfoElement != null)
+				syntaxArray.add(classInfoElement);
+		});
+		return syntaxArray;
+	}
+
+	private JsonObject generateFunctionElement(JavaFunction<?> function) {
+		JsonObject functionJsonObject = new JsonObject();
+		functionJsonObject.addProperty("id", idProvider.getId(function));
+		functionJsonObject.addProperty("name", function.getName());
+		functionJsonObject.addProperty("since", function.getSince());
+		functionJsonObject.add("description", convertToJsonArray(function.getDescription()));
+		functionJsonObject.add("examples", convertToJsonArray(function.getExamples()));
+
+		ClassInfo<?> returnType = function.getReturnType();
+		if (returnType != null) {
+			functionJsonObject.addProperty("return-type", returnType.getDocName());
+		}
+
+		String functionSignature = function.getSignature().toString(true, false);
+		functionJsonObject.add("patterns", convertToJsonArray(new String[] { functionSignature }));
```

Comment:
Should we remove parse tags from json or keep it for other docs websites to handle them as we wish? maybe we can add an option to remove them to make it easier
![image](https://github.com/SkriptLang/Skript/assets/20037329/a6bdffe9-75a6-4816-9ad0-a22c13079551)

Reply:
i think we can just keep them as is. in the future maybe we can expose another field with simplified patterns

Reply:
I like the idea of simplified patterns, good to have it. Another PR is suitable for it 👍